File: spring-boot-oauth2-social-login/src/main/java/com/javachinna/model/Comments.java
package com.javachinna.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
public class Comments {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    private long id;
    @JsonProperty
    private String comment;
    @ManyToOne
    @JsonIgnore
    private User user;
    private int topicId;
    private int tutorialId;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy - HH:mm ")
    @CreationTimestamp
    private LocalDateTime timestamp;

    @JsonGetter("userName")
    public String getUserName(){
        return user.getDisplayName();
    }
}

File: angular-11-social-login/src/app/model/Comments.ts

export class Comments{
    constructor(comment?: Comments){
        this.userName = comment && comment.userName || "";
        this.timestamp = comment && comment.timestamp || "";
        this.comment = comment && comment.userName || "";
        this.topicId = comment && comment.topicId || 0;
        this.tutorialId = comment && comment.tutorialId || 0;
    }


    public userName: string;
    public timestamp: string;
    public comment: string;
    public topicId: number;
    public tutorialId: number;

    setComment(comment: string){
        this.comment = comment;
    }

    setTopicId(topicId: number){
        this.topicId = topicId;
    }

    setTutorialId(tutorialId: number){
        this.tutorialId = tutorialId;
    }
}